### ISSUE TITLE
ClassNotFoundException: androidx.core.util.Preconditions

### ISSUE THREAD

**Issue by brewin** [mopub/mopub-android-mediation#350]
1. **What is the issue?**
    > ClassNotFoundException: androidx.core.util.Preconditions

2. **Check the applicable box(es) below by placing an `x` in the brackets:**
    - [x] I searched https://developers.mopub.com/ and found no answer.
    - [x] I searched https://twittercommunity.com/c/advertiser-api/mopub to make sure that this issue has not already been filed.
    - [x] I searched the adapter changelogs to make sure this issue has not already been resolved.

3. **Fill in the below template as detailed as possible:**
    * **Ad Format** (*Banner, Interstitial, Rewarded Video, Native Static, Native Video*)
        > Banner
    * **Platform** (*Android, iOS, Unity*)
        > Android
    * **MoPub SDK Version**
        > 5.17.0
    * **Mediation SDK Name & Version**
        > Facebook 6.5.0
    * **Mediation Adapter Version**
        > 6.5.0.0
    * **MoPub Ad Unit ID(s) used to reproduce the issue**
        > Provide the data here
    * **Device model and OS Version**
        > Many devices running Android 10/11
    * **Steps to reproduce the issue** (include integration steps if not standard)
        > I updated from Facebook SDK version 6.4.0 to 6.5.0. I assume this is a Proguard/R8 issue.
    * **Observed behavior**
        > Exception and crash
    * **Expected behavior** (if known)
        > Provide the data here
    * **Evidence** (if available)
```
java.lang.ClassNotFoundException: androidx.core.util.Preconditions
    com.facebook.ads.redexgen.X.da.A05
    com.facebook.ads.redexgen.X.da.A09
    com.facebook.ads.redexgen.X.G2.onStop
    com.facebook.ads.redexgen.X.dh.A01
    com.facebook.ads.redexgen.X.UC.onViewDetachedFromWindow
    android.view.View.dispatchDetachedFromWindow View.java:21375
    android.view.ViewGroup.dispatchDetachedFromWindow ViewGroup.java:4649
    android.view.ViewGroup.dispatchDetachedFromWindow ViewGroup.java:4641
    android.view.ViewGroup.removeViewInternal ViewGroup.java:6287
    android.view.ViewGroup.removeViewInternal ViewGroup.java:6258
    android.view.ViewGroup.removeView ViewGroup.java:6189
    com.mopub.common.util.Views.removeFromParent Views.java:3
    com.mopub.mobileads.FacebookBanner.onInvalidate FacebookBanner.java:2
    com.mopub.mobileads.InlineAdAdapter.b InlineAdAdapter.java:2
    com.mopub.mobileads.AdAdapter.d AdAdapter.java:1
    com.mopub.mobileads.AdViewController.d AdViewController.java:2
    com.mopub.mobileads.AdViewController$a.onResponse
    com.mopub.mobileads.AdViewController$a.onResponse
    com.mopub.network.AdLoader.b AdLoader.java:11
    com.mopub.network.AdLoader$a.onResponse
    com.mopub.network.AdLoader$a.onResponse
    com.mopub.network.MultiAdRequest.a MultiAdRequest.java:3
    com.mopub.network.MoPubRequest$1.a MoPubRequest.kt:1
    com.mopub.volley.ExecutorDelivery$b.run
    android.os.Handler.handleCallback Handler.java:883
    android.os.Handler.dispatchMessage Handler.java:100
    android.os.Looper.loop Looper.java:237
    android.app.ActivityThread.main ActivityThread.java:8167
    java.lang.reflect.Method.invoke Method.java
    com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:496
    com.android.internal.os.ZygoteInit.main ZygoteInit.java:1100
```

**Comment by chauduyphanvu** [comment]
@brewin, it is possible Facebook might have introduced a dependency on `androidx.core.util.Preconditions` in the latest SDK release. Do you have an exclusion for that package in your Proguard already?

Vu Chau
MoPub Team

**Comment by brewin** [comment]
I don't have anything related to any androidx package or class in my proguard rules. I reported this to Facebook, but thought I'd report it here too in case anyone else runs into this issue. I haven't heard anything back from Facebook.

For now, I've disabled Facebook ads and downgraded to 6.4.0 since I don't know for sure whether just adding a `-keep class androidx.core.util.Preconditions` will fix it.

**Comment by chauduyphanvu** [comment]
@brewin, Facebook confirmed with us that this was a backend bug on their side and should be resolved via the upcoming Audience Network SDK release. Stay tuned!

**Comment by brewin** [comment]
@chauduyphanvu Do you know whether this was fixed in 6.5.1? It's not mentioned in the [changelog](https://developers.facebook.com/docs/audience-network/setting-up/platform-setup/android/changelog/).

**Comment by chauduyphanvu** [comment]
@brewin, I am checking with our Facebook contacts on whether they can provide a confirmation. Will post an update when I hear back from them.

**Comment by chauduyphanvu** [comment]
@brewin, Facebook confirmed that this issue was fixed in the 6.5.1 [release](https://developers.facebook.com/docs/audience-network/setting-up/platform-setup/android/changelog). However, the feature was turned off in Android for versions 6.5.0 and lower so it shouldn't affect pubs who haven't upgraded.
